packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -1423,10 +1442,14 @@ export function attach(
       }
     }
 
-    fiberToFiberInstanceMap.delete(fiber);
+    if (fiberToFiberInstanceMap.get(fiber) === fiberInstance) {
```

Comment:
Reparenting can happen in different sequences.

E.g. if it moves to later then it can get removed first and then added. In that case we end up with the situation where we have an alternate but is actually a mount.

If it moves to earlier then it gets added first and then later removed. In this case we steal/claim the `fiberToFiberInstanceMap` slot for this Fiber using the newly mounted instance. Later the previous instance gets removed which shouldn't delete the newly mounted instance from this map.